zeus/utils/framework.py

```diff
@@ -105,41 +105,43 @@ def sync_execution(
 
 
 def all_reduce(
-    object: List[int] | List[float], operation: Literal["sum", "max"]
-) -> int | float:
+    object: list[int] | list[float], operation: Literal["sum", "max"]
+) -> list[int] | list[float]:
     """Reduce objects from all replicas through the specified operation.
 
-    If running in a distributed setting, the objects are reduced across all replicas.
-    If running in a non-distributed setting, the operation is just done on the single object.
+    If the current execution is not distributed, the object is returned as is.
     """
     if torch_is_available(ensure_cuda=False):
         torch = MODULE_CACHE["torch"]
 
-        # wrap object in a tensor if it is not already
-        if not isinstance(object, torch.Tensor):
-            object = torch.Tensor(object)
+        # if torch.distributed is not available or not initialized, return the object as is
+        if (
+            not torch.distributed.is_available()
+            or not torch.distributed.is_initialized()
+        ):
+            return object
+
+        # wrap object in a tensor
+        tensor = torch.Tensor(object).cuda()
 
         # determine operation
         if operation == "sum":
-            torch_func = torch.sum
             torch_op = torch.distributed.ReduceOp.SUM
         elif operation == "max":
-            torch_func = torch.max
             torch_op = torch.distributed.ReduceOp.MAX
         else:
             raise ValueError(f"all_reduce unsupported operation: {operation}")
 
-        # compute local operation
-        result = torch_func(object)
-
-        # all-reduce only if torch.distributed is available and initialized
-        if torch.distributed.is_available() and torch.distributed.is_initialized():
-            torch.distributed.all_reduce(result.cuda(), op=torch_op)
-        return result.item()
+        torch.distributed.all_reduce(tensor, op=torch_op)
+        return tensor.cpu().tolist()
 
     if jax_is_available():
-        # JAX cross-device all-reduce not yet implemente
-        return sum(object) if operation == "sum" else max(object)
+        # Check if not distributed
+        jax = MODULE_CACHE["jax"]
+        if jax.process_count() == 1:
```

Comment:
https://jax.readthedocs.io/en/latest/multi_process.html#running-multi-process-computations
Should be `jax.device_count()`?

Reply:
yes, it should be. Fixed that